### ISSUE TITLE
test_generate_gir_with_address_sanitizer fails in Gentoo under our package manager testing

### ISSUE THREAD

**Issue by williamh** [mesonbuild/meson#6486]
See this downstream bug report. 

https://bugs.gentoo.org/673016

Our package manager uses sandboxing, but this causes the given test to fail. Currently we disable the test by running this command inside our build:

```
sed -i -e 's/test_generate_gir_with_address_sanitizer/_&/' run_unittests.py
```

I'm filing this issue because I prefer to not change code downstream if I can avoid doing so.
I want to know if there is a better way we can handle this situation.
Is there some way you can help us with  this?

Thanks for your time. I'm tagging my co-maintainer also so he can follow along.

cc: @floppym

**Comment by bonzini** [comment]
@williamh can you check this with meson.git?

**Comment by thesamesam** [comment]
With my Gentoo hat on too, this isn't really an upstream bug unless we can think of a clean way of doing this like skipping if `LD_PRELOAD` is set in the environment (our sandboxing also uses `LD_PRELOAD` which conflicts with ASAN's "am I first" check).

I guess I could add that in, so mine.

**Comment by bonzini** [comment]
I agree but Meson switched to checking sanitizers with a compile/link test. This should either fix it, or make it possible to detect the situation and skip the test.

**Comment by thesamesam** [comment]
It's a runtime problem, though?

**Comment by bonzini** [comment]
If it helps improving the detection, it's possible to add a runtime test to the existing compiler and linker test.
